File: 算法/代码随想录算法PDF/00调试源代码/349.两个数组的交集.py
#
# @lc app=leetcode.cn id=349 lang=python3
#
# [349] 两个数组的交集
#

# @lc code=start
class Solution:
    def intersection(self, nums1: List[int], nums2: List[int]) -> List[int]:
        resSet = set()
        numSet = set()
        for num in nums1:
            numSet.add(num)
        for num in nums2:
            if num in numSet:
                resSet.add(num)
        return list(resSet)
# @lc code=end

File: 算法/代码随想录算法PDF/00调试源代码/349.两个数组的交集.cpp
/*
 * @lc app=leetcode.cn id=349 lang=cpp
 *
 * [349] 两个数组的交集
 */

// @lc code=start
class Solution {
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) {
        unordered_set<int> resSet;
        vector<int> numVec(1001, 0);
        for (int num : nums1) {
            numVec[num] = 1;
        }
        for (int num : nums2) {
            if (numVec[num] == 1) {
                resSet.insert(num);
            }
        }
        return vector<int> (resSet.begin(), resSet.end());
    }
};
// @lc code=end

